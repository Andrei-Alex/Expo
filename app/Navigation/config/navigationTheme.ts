import { DefaultTheme } from '@react-navigation/native';
import { colors } from '../../config';

export const appTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: colors.primary,
    background: colors.white
  }
};
