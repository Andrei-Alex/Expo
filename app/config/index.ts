export { default as colors } from './theme/colors';
