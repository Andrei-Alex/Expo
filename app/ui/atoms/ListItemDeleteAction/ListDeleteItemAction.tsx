import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { styles, Props } from '.';
import { colors } from '../../../config';

const ListDeleteItemAction: React.FC<Props> = ({ style, onPress }) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={[styles.container, style]}>
        <MaterialCommunityIcons
          name={'trash-can'}
          size={35}
          color={colors.white}
        />
      </View>
    </TouchableWithoutFeedback>
  );
};

export default ListDeleteItemAction;
