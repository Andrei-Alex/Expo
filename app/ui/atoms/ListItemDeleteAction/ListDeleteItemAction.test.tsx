import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import renderer from 'react-test-renderer';
import ListDeleteItemAction from './ListDeleteItemAction';
import { colors } from '../../../config';

describe('ListDeleteItemAction', () => {
  let component;
  const onPressMock = jest.fn();
  const styles = { backgroundColor: colors.danger };

  beforeEach(() => {
    onPressMock.mockReset();
    component = renderer.create(
      <ListDeleteItemAction style={styles} onPress={onPressMock} />
    );
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('receives custom styles correctly', () => {
    const instance = component.root;
    const viewComponent = instance.findByType(View);

    const receivedStyle = viewComponent.props.style;
    const styleArray = Array.isArray(receivedStyle)
      ? receivedStyle
      : [receivedStyle];

    const hasTestStyle = styleArray.some((style) =>
      Object.entries(styles).every(([key, value]) => style[key] === value)
    );
    expect(hasTestStyle).toBeTruthy();
  });

  it('triggers onPress when pressed', () => {
    const touchable = component.root.findByType(TouchableWithoutFeedback);
    touchable.props.onPress();

    expect(onPressMock).toHaveBeenCalled();
  });
});
