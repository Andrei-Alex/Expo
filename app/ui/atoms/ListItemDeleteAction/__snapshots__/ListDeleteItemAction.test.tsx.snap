// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDeleteItemAction renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#ff5252",
        "justifyContent": "center",
        "width": 70,
      },
      {
        "backgroundColor": "#ff5252",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#fff",
          "fontSize": 35,
        },
        undefined,
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ó°©¹
  </Text>
</View>
`;
